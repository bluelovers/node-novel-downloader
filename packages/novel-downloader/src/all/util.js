"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isIDKEY = exports.isSiteID = exports.IDKEY2siteID = exports.siteID2IDKEY = void 0;
/**
 * Created by user on 2020/2/3.
 */
const const_1 = require("./const");
function siteID2IDKEY(siteID) {
    if (isIDKEY(siteID)) {
        return const_1.EnumIDKEYList[siteID];
    }
}
exports.siteID2IDKEY = siteID2IDKEY;
function IDKEY2siteID(input) {
    if (!isIDKEY(input)) {
        input = siteID2IDKEY(input);
    }
    if (isIDKEY(input)) {
        return const_1.EnumIDKEYToSiteID[input];
    }
}
exports.IDKEY2siteID = IDKEY2siteID;
function isSiteID(siteID) {
    return (siteID in const_1.EnumNovelSiteList) && typeof const_1.EnumNovelSiteList[siteID] === 'string';
}
exports.isSiteID = isSiteID;
function isIDKEY(IDKEY) {
    return (IDKEY in const_1.EnumIDKEYList) && typeof const_1.EnumIDKEYList[IDKEY] === 'string';
}
exports.isIDKEY = isIDKEY;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoidXRpbC5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbInV0aWwudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6Ijs7O0FBQUE7O0dBRUc7QUFDSCxtQ0FBOEU7QUFFOUUsU0FBZ0IsWUFBWSxDQUFDLE1BQWtEO0lBRTlFLElBQUksT0FBTyxDQUFDLE1BQU0sQ0FBQyxFQUNuQjtRQUNDLE9BQU8scUJBQWEsQ0FBQyxNQUFNLENBQUMsQ0FBQTtLQUM1QjtBQUNGLENBQUM7QUFORCxvQ0FNQztBQUVELFNBQWdCLFlBQVksQ0FBQyxLQUFpRDtJQUU3RSxJQUFJLENBQUMsT0FBTyxDQUFDLEtBQUssQ0FBQyxFQUNuQjtRQUNDLEtBQUssR0FBRyxZQUFZLENBQUMsS0FBSyxDQUFDLENBQUM7S0FDNUI7SUFFRCxJQUFJLE9BQU8sQ0FBQyxLQUFLLENBQUMsRUFDbEI7UUFDQyxPQUFPLHlCQUFpQixDQUFDLEtBQVksQ0FBQyxDQUFBO0tBQ3RDO0FBQ0YsQ0FBQztBQVhELG9DQVdDO0FBRUQsU0FBZ0IsUUFBUSxDQUFDLE1BQWtDO0lBRTFELE9BQU8sQ0FBQyxNQUFNLElBQUkseUJBQWlCLENBQUMsSUFBSSxPQUFPLHlCQUFpQixDQUFDLE1BQU0sQ0FBQyxLQUFLLFFBQVEsQ0FBQTtBQUN0RixDQUFDO0FBSEQsNEJBR0M7QUFFRCxTQUFnQixPQUFPLENBQUMsS0FBNkI7SUFFcEQsT0FBTyxDQUFDLEtBQUssSUFBSSxxQkFBYSxDQUFDLElBQUksT0FBTyxxQkFBYSxDQUFDLEtBQUssQ0FBQyxLQUFLLFFBQVEsQ0FBQTtBQUM1RSxDQUFDO0FBSEQsMEJBR0MiLCJzb3VyY2VzQ29udGVudCI6WyIvKipcbiAqIENyZWF0ZWQgYnkgdXNlciBvbiAyMDIwLzIvMy5cbiAqL1xuaW1wb3J0IHsgRW51bU5vdmVsU2l0ZUxpc3QsIEVudW1JREtFWUxpc3QsIEVudW1JREtFWVRvU2l0ZUlEIH0gZnJvbSAnLi9jb25zdCc7XG5cbmV4cG9ydCBmdW5jdGlvbiBzaXRlSUQySURLRVkoc2l0ZUlEOiBFbnVtTm92ZWxTaXRlTGlzdCB8IHN0cmluZyB8IEVudW1JREtFWUxpc3QpOiBFbnVtSURLRVlMaXN0XG57XG5cdGlmIChpc0lES0VZKHNpdGVJRCkpXG5cdHtcblx0XHRyZXR1cm4gRW51bUlES0VZTGlzdFtzaXRlSURdXG5cdH1cbn1cblxuZXhwb3J0IGZ1bmN0aW9uIElES0VZMnNpdGVJRChpbnB1dDogRW51bU5vdmVsU2l0ZUxpc3QgfCBzdHJpbmcgfCBFbnVtSURLRVlMaXN0KTogRW51bU5vdmVsU2l0ZUxpc3Rcbntcblx0aWYgKCFpc0lES0VZKGlucHV0KSlcblx0e1xuXHRcdGlucHV0ID0gc2l0ZUlEMklES0VZKGlucHV0KTtcblx0fVxuXG5cdGlmIChpc0lES0VZKGlucHV0KSlcblx0e1xuXHRcdHJldHVybiBFbnVtSURLRVlUb1NpdGVJRFtpbnB1dCBhcyBhbnldXG5cdH1cbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGlzU2l0ZUlEKHNpdGVJRDogRW51bU5vdmVsU2l0ZUxpc3QgfCBzdHJpbmcpOiBzaXRlSUQgaXMgRW51bU5vdmVsU2l0ZUxpc3Rcbntcblx0cmV0dXJuIChzaXRlSUQgaW4gRW51bU5vdmVsU2l0ZUxpc3QpICYmIHR5cGVvZiBFbnVtTm92ZWxTaXRlTGlzdFtzaXRlSURdID09PSAnc3RyaW5nJ1xufVxuXG5leHBvcnQgZnVuY3Rpb24gaXNJREtFWShJREtFWTogRW51bUlES0VZTGlzdCB8IHN0cmluZyk6IElES0VZIGlzIEVudW1JREtFWUxpc3Rcbntcblx0cmV0dXJuIChJREtFWSBpbiBFbnVtSURLRVlMaXN0KSAmJiB0eXBlb2YgRW51bUlES0VZTGlzdFtJREtFWV0gPT09ICdzdHJpbmcnXG59XG4iXX0=