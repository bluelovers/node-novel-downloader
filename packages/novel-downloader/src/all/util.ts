/**
 * Created by user on 2020/2/3.
 */
import { EnumNovelSiteList, EnumIDKEYList, EnumIDKEYToSiteID } from './const';

export function siteID2IDKEY(siteID: EnumNovelSiteList | string | EnumIDKEYList): EnumIDKEYList
{
	if (isIDKEY(siteID))
	{
		return EnumIDKEYList[siteID]
	}
}

export function IDKEY2siteID(input: EnumNovelSiteList | string | EnumIDKEYList): EnumNovelSiteList
{
	if (!isIDKEY(input))
	{
		input = siteID2IDKEY(input);
	}

	if (isIDKEY(input))
	{
		return EnumIDKEYToSiteID[input as any]
	}
}

export function isSiteID(siteID: EnumNovelSiteList | string): siteID is EnumNovelSiteList
{
	return (siteID in EnumNovelSiteList) && typeof EnumNovelSiteList[siteID] === 'string'
}

export function isIDKEY(IDKEY: EnumIDKEYList | string): IDKEY is EnumIDKEYList
{
	return (IDKEY in EnumIDKEYList) && typeof EnumIDKEYList[IDKEY] === 'string'
}
