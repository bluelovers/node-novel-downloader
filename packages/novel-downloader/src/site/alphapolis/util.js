"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.parseUrl = exports.makeUrl = exports.check = void 0;
const url_1 = __importStar(require("../../util/url"));
function check(url, options) {
    return /alphapolis/i.test(url_1.default(url).hostname || '');
}
exports.check = check;
function makeUrl(urlobj, bool, ...argv) {
    let url;
    let cid = (!bool && urlobj.chapter_id) ? `episode\/${urlobj.chapter_id}` : '';
    url = `https://www.alphapolis.co.jp/novel/${urlobj.novel_pid}/${urlobj.novel_id}/${cid}`;
    return url_1.default(url);
}
exports.makeUrl = makeUrl;
function parseUrl(_url, ...argv) {
    const { urlobj, url } = url_1._handleParseURL(_url, ...argv);
    let r;
    let m;
    r = /alphapolis\.co\.jp\/novel\/([^\/]+)\/([^\/]+)(?:\/episode\/([^\/]+))?/;
    m = r.exec(url);
    if (!m) {
        r = /novel\/([^\/]+)\/([^\/]+)(?:\/episode\/([^\/]+))?/;
        m = r.exec(url);
    }
    if (m) {
        urlobj.novel_pid = m[1];
        urlobj.novel_id = m[2];
        urlobj.chapter_id = m[3];
        return urlobj;
    }
    return urlobj;
}
exports.parseUrl = parseUrl;
//# sourceMappingURL=util.js.map