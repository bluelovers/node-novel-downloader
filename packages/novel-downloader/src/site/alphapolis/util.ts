import type NovelSite from '../index';
import createURL, { _handleParseURL, IParseUrlRuntime } from '../../util/url';

export function check(url: string | URL | NovelSite.IParseUrl, options?): boolean
{
	return /alphapolis/i.test(createURL(url as any).hostname || '');
}

export function makeUrl(urlobj: NovelSite.IParseUrl, bool ?: boolean | number, ...argv)
{
	let url: string;

	let cid = (!bool && urlobj.chapter_id) ? `episode\/${urlobj.chapter_id}` : '';

	url = `https://www.alphapolis.co.jp/novel/${urlobj.novel_pid}/${urlobj.novel_id}/${cid}`;

	return createURL(url);
}

export function parseUrl(_url: string | URL | number, ...argv)
{
	const { urlobj, url } = _handleParseURL(_url, ...argv);

	let r: RegExp;
	let m: RegExpExecArray;

	r = /alphapolis\.co\.jp\/novel\/([^\/]+)\/([^\/]+)(?:\/episode\/([^\/]+))?/;
	m = r.exec(url as string);

	if (!m)
	{
		r = /novel\/([^\/]+)\/([^\/]+)(?:\/episode\/([^\/]+))?/;
		m = r.exec(url as string)
	}

	if (m)
	{
		urlobj.novel_pid = m[1];
		urlobj.novel_id = m[2];
		urlobj.chapter_id = m[3];

		return urlobj;
	}

	return urlobj;
}
