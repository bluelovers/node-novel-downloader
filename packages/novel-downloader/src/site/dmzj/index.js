"use strict";
/**
 * Created by user on 2017/12/6/006.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NovelSiteDmzj = void 0;
const tslib_1 = require("tslib");
const fs_extra_1 = (0, tslib_1.__importDefault)(require("fs-extra"));
const util_1 = require("fs-iconv/util");
const upath2_1 = (0, tslib_1.__importDefault)(require("upath2"));
const node_novel_info_1 = (0, tslib_1.__importDefault)(require("node-novel-info"));
const jsdom_extra_1 = require("jsdom-extra");
const index_1 = (0, tslib_1.__importStar)(require("../index"));
const index_2 = require("../index");
const index_3 = require("../index");
const shortid_1 = (0, tslib_1.__importDefault)(require("shortid"));
const fetch_1 = require("../../fetch");
const jsdom_1 = require("../../jsdom");
/**
 * @deprecated
 */
// @ts-ignore
let NovelSiteDmzj = class NovelSiteDmzj extends index_1.default {
    _makeUrl(urlobj, page) {
        if (!urlobj.chapter_id) {
            if (!urlobj.volume_id) {
                return `http://q.dmzj.com/${urlobj.novel_id}/index.shtml`;
            }
            return `http://q.dmzj.com/${urlobj.novel_id}/${urlobj.volume_id}/${urlobj.volume_id}.txt`;
        }
        return `http://q.dmzj.com/${urlobj.novel_id}/${urlobj.volume_id}/${urlobj.chapter_id}${page
            ? '_' + page
            : ''}.shtml`;
    }
    makeUrl(urlobj, bool) {
        let url = this._makeUrl(urlobj, bool);
        // @ts-ignore
        return new URL(url);
    }
    parseUrl(url) {
        let urlobj = {
            // @ts-ignore
            url: new URL(url),
            novel_id: null,
            volume_id: null,
            chapter_id: null,
        };
        url = urlobj.url.href;
        let r = /(?:q\.dmzj\.com\/|^\/)(?:(\d+)\/(?:(\d+)\/(?:(\d+)[\._])?)?)/;
        let m = r.exec(url.toString());
        if (m) {
            urlobj.novel_id = m[1];
            urlobj.volume_id = m[2];
            urlobj.chapter_id = m[3];
        }
        // @ts-ignore
        return urlobj;
    }
    // @ts-ignore
    download(url, downloadOptions = {}) {
        const self = this;
        const [PATH_NOVEL_MAIN, optionsRuntime] = this.getOutputDir(downloadOptions);
        /*
        optionsRuntime.optionsJSDOM = Object.assign({}, defaultJSDOMOptions, optionsRuntime.optionsJSDOM, {
            runScripts: 'dangerously',
        });
        */
        optionsRuntime.optionsJSDOM = (0, jsdom_1.createOptionsJSDOM)(optionsRuntime.optionsJSDOM, {
            runScripts: 'dangerously',
        });
        let path_main;
        return index_2.PromiseBluebird
            .bind(self)
            .then(async function () {
            let _data = await self._download_info(url, optionsRuntime);
            path_main = upath2_1.default.join(self.PATH_NOVEL_MAIN, (0, util_1.trimFilename)(`${_data.data.g_lnovel_name}_(${_data.data.g_lnovel_id})`));
            let _a = _data.value.reduce(function (a, b) {
                return a.concat(b.chapter);
            }, []);
            let _f = await index_2.PromiseBluebird
                .mapSeries(_a, function (a, index, len) {
                let pad_len = len.toString().length > 4 ? len.toString().length : 4;
                // @ts-ignore
                let volume_name = a.volume_name;
                // @ts-ignore
                console.log(a.url);
                // @ts-ignore
                return self._downloadChapter(a.url, optionsRuntime)
                    .then(async function (data) {
                    let _file = upath2_1.default.join(path_main, 
                    // @ts-ignore
                    (0, util_1.trimFilename)(`${a.index_volume.toString()
                        .padStart(pad_len, '0')} ${volume_name}_(${data.data.g_volume_id})`), 
                    // @ts-ignore
                    (0, util_1.trimFilename)(`${a.index.toString()
                        .padStart(pad_len, '0')}_${data.data.chapter_name}.${data.data.g_chapter_id}`));
                    await fs_extra_1.default.outputJson(_file + '.json', data, {
                        spaces: "\t",
                    });
                    await fs_extra_1.default.outputFile(_file + '.txt', data.value);
                    return upath2_1.default.relative(self.PATH_NOVEL_MAIN, _file);
                })
                    .then(function (file) {
                    console.log('done', file);
                    return file;
                });
            })
                .then(function (ret) {
                //console.log('done', ret);
                return ret;
            });
            // @ts-ignore
            _data.checkdate = (0, index_3.moment)().tz(index_3.moment.tz.guess());
            // @ts-ignore
            _data.files = _f;
            return _data;
        })
            .tap(async function (novel) {
            await fs_extra_1.default.outputJson(upath2_1.default.join(path_main, (0, util_1.trimFilename)(`${novel.data.g_lnovel_name}.${novel.data.g_lnovel_id}`)) + '.json', novel, {
                spaces: "\t",
            });
            let options = {};
            options[self.IDKEY] = {
            //txtdownload_id: novel.novel_syosetu_id,
            };
            let md = node_novel_info_1.default.stringify({
                novel: {
                    tags: [
                        self.IDKEY,
                    ],
                    series: {
                        // @ts-ignore
                        name: novel.novel_series_title || '',
                    },
                },
                options,
                // @ts-ignore
                link: novel.link || [],
            }, novel, {
                options: {
                    textlayout: {
                        allow_lf2: true,
                    }
                },
            });
            let file = upath2_1.default.join(path_main, `README.md`);
            await fs_extra_1.default.outputFile(file, md);
        });
    }
    async _download_info(url, optionsRuntime = {}) {
        const self = this;
        {
            let data = self.parseUrl(url);
            if (!data.novel_id) {
                throw new ReferenceError();
            }
            url = self.makeUrl({
                novel_id: data.novel_id,
            });
        }
        let _data = {
            url: url,
            data: {},
            value: null,
        };
        return await (0, jsdom_extra_1.fromURL)(url, optionsRuntime.optionsJSDOM)
            .then(function (dom) {
            const $ = dom.$;
            // @ts-ignore
            const window = dom.window;
            //console.log(dom.serialize());
            _data.data.g_lnovel_id = window.g_lnovel_id;
            _data.data.g_lnovel_name = window.g_lnovel_name;
            _data.data.cover_pic = $('#cover_pic').attr('src');
            {
                let _t = $('.main .pic .con').text();
                _data.data.con = _t;
                if (_t.match(/作者：(.+)/)) {
                    _data.data.author = RegExp.$1;
                }
                if (_t.match(/类型：(.+)/)) {
                    _data.data.type = (RegExp.$1).toString().split('/');
                }
                if (_t.match(/状态：(.+)/)) {
                    _data.data.status = RegExp.$1;
                }
                if (_t.match(/更新：(.+)/)) {
                    _data.data.lastupdate = index_3.moment.tz(RegExp.$1, 'YYYY-MM-DD HH:mm:ss', 'Asia/Taipei');
                }
            }
            // @ts-ignore
            _data.data.desc = $('#detail_block > .ilist:has(> h3) > p').text();
            window.volume_list = window.volume_list.reverse();
            window.chapter_list = window.chapter_list.reverse();
            let list = [];
            for (let v in window.chapter_list) {
                let volume_name = $(window.volume_list[v]).find('.chapnamesub').text();
                window.chapter_list[v] = window.chapter_list[v].reverse();
                //let data = parseUrl($(dom.window.chapter_list[v][0]).attr('href'));
                list[v] = {
                    index: v,
                    novel_id: null,
                    volume_name: volume_name,
                    volume_id: null,
                    chapter: [],
                };
                for (let ci in window.chapter_list[v]) {
                    if (!window.chapter_list[v][ci]) {
                        continue;
                    }
                    let _a = $(window.chapter_list[v][ci]);
                    //console.log(_a);
                    let _d = self.parseUrl(_a.prop('href'));
                    if (!list[v].volume_id) {
                        list[v].novel_id = _d.novel_id;
                        list[v].volume_id = _d.volume_id;
                    }
                    list[v].chapter.push({
                        index: ci,
                        index_volume: v,
                        novel_id: _d.novel_id,
                        volume_id: _d.volume_id,
                        volume_name: volume_name,
                        chapter_name: _a.text(),
                        chapter_id: _d.chapter_id,
                        url: _a.prop('href'),
                    });
                }
            }
            _data.value = list;
            return _data;
        })
            .then(function (_data) {
            // @ts-ignore
            _data.checkdate = (0, index_3.moment)().tz(index_3.moment.tz.guess());
            return _data;
        });
    }
    async _downloadChapter(data, optionsRuntime) {
        const self = this;
        let url;
        {
            data = self.parseUrl(data);
            if (!data.novel_id) {
                throw new ReferenceError();
            }
            url = self.makeUrl(data);
        }
        let _data = {};
        let $;
        return (0, jsdom_extra_1.fromURL)(url, optionsRuntime.optionsJSDOM)
            .then(async (dom) => {
            let window = dom.window;
            $ = dom.$;
            let chapter_name = window.g_chapter_name
                .replace(/\\/ig, '')
                .replace(/^[\s\uFEFF\xA0　]+|[\s\uFEFF\xA0　]+$/g, '');
            let volume_name = window.document
                .querySelector('#page_contents .tit')
                .innerHTML
                .replace(/\\/ig, '')
                .replace(chapter_name, '')
                .replace(/^[\s\uFEFF\xA0　]+|[\s\uFEFF\xA0　]+$/g, '');
            _data = {
                url: url,
                data: {
                    g_lnovel_id: window.g_lnovel_id,
                    g_volume_id: window.g_volume_id,
                    g_chapter_id: window.g_chapter_id,
                    g_lnovel_name: window.g_lnovel_name,
                    g_volume_name: volume_name,
                    g_chapter_name: chapter_name,
                    chapter_name: chapter_name,
                    volume_name: volume_name,
                },
                value: null,
            };
            let pa = [];
            pa[0] = window.document.querySelector("#chapter_contents_first").innerHTML;
            if (window.g_chapter_pages_count > 1) {
                let ua = window.g_chapter_pages_url
                    .filter(function (value) {
                    return value;
                })
                    .map(function (value, index, array) {
                    if (value) {
                        value = 'http://q.dmzj.com/' + value;
                    }
                    return value;
                });
                await (0, fetch_1.manyRequest)(ua, {
                    // @ts-ignore
                    encoding: null,
                    resolveWithFullResponse: true,
                })
                    .then(async function (ret) {
                    pa = pa.concat(ret.map(function (value, index, array) {
                        //console.log(value.body);
                        return value.body.toString();
                    }));
                });
            }
            return pa;
        })
            .then(async function (pa) {
            _data.imgs = [];
            pa = await index_2.PromiseBluebird.mapSeries(pa, async function (value, idx) {
                let _c = {};
                //let $ = cheerio.load(value);
                let _a = $(value);
                let _img = _a.find('img');
                //console.log(_img.length);
                if (_img.length) {
                    _img.each(function (index, elem) {
                        // @ts-ignore
                        let _this = $(this);
                        if (_this.prop('src')) {
                            let id = (0, shortid_1.default)();
                            _c[id] = _this.prop('src');
                            _data.imgs.push(_c[id]);
                            // @ts-ignore
                            $(`<span>{{@${id}@}}</span>`).insertAfter(this);
                            // @ts-ignore
                            $(this)
                                .remove();
                        }
                    });
                    //console.log($.html());
                }
                let _t = _a.text();
                for (let id in _c) {
                    /*
                    if (!/^(?:[a-z]\:|\:)?\/\//i.test(_c[id]))
                    {
                        _c[id] = 'http://q.dmzj.com/' + _c[id];
                    }
                    */
                    /*
                    await download_image(_c[id], {
                        fromfile: file,
                        prefix: 'img_' + (_idx++).toString().padStart(3, '0') + '_',
                    });
                    */
                    _t = _t.replace(`{{@${id}@}}`, `\n\n<img src="${_c[id]}"/>\n\n`);
                }
                return _t;
            });
            return pa.join("\n");
        })
            .then(function (html) {
            return html
                .replace(/^\s*(?:<p>)?/i, '')
                .replace(/\s*<(?:\/?p|br\/?)>\s*$/i, '')
                .replace(/\r\n|\r(?!\n)/g, "\n")
                //.replace(/<\/p><p>\n/ig, "\n")
                //.replace(/<p><\/p>/ig, "\n")
                //.replace(/(<\/p>|<p>)\n/ig, "\n")
                .replace(/[\t\uFEFF\xA0　]+(\n|$)/ig, "$1")
                .replace(/(\n)[\t]+/ig, "$1")
                .replace(/\s+$/ig, "")
                .replace(/\n{3,}/ig, "\n\n");
        })
            .then(function (html) {
            _data.value = html;
            _data.checkdate = (0, index_3.moment)().tz(index_3.moment.tz.guess());
            return _data;
        });
    }
};
NovelSiteDmzj.disabled = true;
NovelSiteDmzj.IDKEY = 'dmzj';
NovelSiteDmzj = (0, tslib_1.__decorate)([
    (0, index_1.staticImplements)()
], NovelSiteDmzj);
exports.NovelSiteDmzj = NovelSiteDmzj;
exports.default = NovelSiteDmzj;
//# sourceMappingURL=index.js.map