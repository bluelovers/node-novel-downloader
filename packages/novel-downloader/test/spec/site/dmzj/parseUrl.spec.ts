import { parseUrl } from '../../../../src/site/dmzj/util';

describe(`support`, () =>
{

	([
		'http://v2.api.dmzj.com/novel/2229.json',

		'http://v2.api.dmzj.com/novel/download/2229_8303_69136.txt',

		'2440',

		1279,

		'http://q.dmzj.com/2476/index.shtml',

		'http://jurisdiction.dmzj1.com/lnovel/7528_62192.txt?t=1625273472314&k=e833282eb316c122004f04a6e3d416b3',

		'http://nnv4api.muwai.com/novel/chapter/784.json',
		'http://nnv4api.muwai.com/novel/chapter/784',
		'http://nnv4api.muwai.com/novel/chapter/784/',

		'http://nnv3api.muwai.com/novel/chapter/784.json',
		'http://nnv3api.muwai.com/novel/chapter/784',
		'http://nnv3api.muwai.com/novel/chapter/784/',

		'http://api.muwai.com/novel/chapter/784.json',
		'http://api.muwai.com/novel/chapter/784',
		'http://api.muwai.com/novel/chapter/784/',

		'http://nnv4api.dmzj4.com/novel/chapter/784.json',
		'http://nnv4api.dmzj4.com/novel/chapter/784',
		'http://nnv4api.dmzj4.com/novel/chapter/784/',

		'http://nnv3api.dmzj4.com/novel/chapter/784.json',
		'http://nnv3api.dmzj4.com/novel/chapter/784',
		'http://nnv3api.dmzj4.com/novel/chapter/784/',

		'http://api.dmzj4.com/novel/chapter/784.json',
		'http://api.dmzj4.com/novel/chapter/784',
		'http://api.dmzj4.com/novel/chapter/784/',

	]).forEach((input) => {

		test(`${input}`, () =>
		{

			let actual = parseUrl(input);

			expect(actual).toHaveProperty('url')
			expect(actual).toHaveProperty('novel_id')
			expect(actual).toHaveProperty('novel_pid')
			expect(actual).toHaveProperty('chapter_id')

			expect(actual).toMatchSnapshot();

		});

	});

})
