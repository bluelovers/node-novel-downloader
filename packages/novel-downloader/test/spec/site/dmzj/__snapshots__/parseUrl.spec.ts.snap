// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`support 1279 1`] = `
{
  "chapter_id": null,
  "novel_id": "1279",
  "novel_pid": null,
  "url": "1279",
}
`;

exports[`support 2440 1`] = `
{
  "chapter_id": null,
  "novel_id": "2440",
  "novel_pid": null,
  "url": "2440",
}
`;

exports[`support http://api.dmzj4.com/novel/chapter/784 1`] = `
{
  "chapter_id": null,
  "novel_id": "784",
  "novel_pid": null,
  "url": "http://api.dmzj4.com/novel/chapter/784",
}
`;

exports[`support http://api.dmzj4.com/novel/chapter/784.json 1`] = `
{
  "chapter_id": null,
  "novel_id": "784",
  "novel_pid": null,
  "url": "http://api.dmzj4.com/novel/chapter/784.json",
}
`;

exports[`support http://api.dmzj4.com/novel/chapter/784/ 1`] = `
{
  "chapter_id": null,
  "novel_id": "784",
  "novel_pid": null,
  "url": "http://api.dmzj4.com/novel/chapter/784/",
}
`;

exports[`support http://api.muwai.com/novel/chapter/784 1`] = `
{
  "chapter_id": null,
  "novel_id": "784",
  "novel_pid": null,
  "url": "http://api.muwai.com/novel/chapter/784",
}
`;

exports[`support http://api.muwai.com/novel/chapter/784.json 1`] = `
{
  "chapter_id": null,
  "novel_id": "784",
  "novel_pid": null,
  "url": "http://api.muwai.com/novel/chapter/784.json",
}
`;

exports[`support http://api.muwai.com/novel/chapter/784/ 1`] = `
{
  "chapter_id": null,
  "novel_id": "784",
  "novel_pid": null,
  "url": "http://api.muwai.com/novel/chapter/784/",
}
`;

exports[`support http://jurisdiction.dmzj1.com/lnovel/7528_62192.txt?t=1625273472314&k=e833282eb316c122004f04a6e3d416b3 1`] = `
{
  "chapter_id": "62192",
  "novel_id": null,
  "novel_pid": null,
  "url": "http://jurisdiction.dmzj1.com/lnovel/7528_62192.txt?t=1625273472314&k=e833282eb316c122004f04a6e3d416b3",
  "volume_id": "7528",
}
`;

exports[`support http://nnv3api.dmzj4.com/novel/chapter/784 1`] = `
{
  "chapter_id": null,
  "novel_id": "784",
  "novel_pid": null,
  "url": "http://nnv3api.dmzj4.com/novel/chapter/784",
}
`;

exports[`support http://nnv3api.dmzj4.com/novel/chapter/784.json 1`] = `
{
  "chapter_id": null,
  "novel_id": "784",
  "novel_pid": null,
  "url": "http://nnv3api.dmzj4.com/novel/chapter/784.json",
}
`;

exports[`support http://nnv3api.dmzj4.com/novel/chapter/784/ 1`] = `
{
  "chapter_id": null,
  "novel_id": "784",
  "novel_pid": null,
  "url": "http://nnv3api.dmzj4.com/novel/chapter/784/",
}
`;

exports[`support http://nnv3api.muwai.com/novel/chapter/784 1`] = `
{
  "chapter_id": null,
  "novel_id": "784",
  "novel_pid": null,
  "url": "http://nnv3api.muwai.com/novel/chapter/784",
}
`;

exports[`support http://nnv3api.muwai.com/novel/chapter/784.json 1`] = `
{
  "chapter_id": null,
  "novel_id": "784",
  "novel_pid": null,
  "url": "http://nnv3api.muwai.com/novel/chapter/784.json",
}
`;

exports[`support http://nnv3api.muwai.com/novel/chapter/784/ 1`] = `
{
  "chapter_id": null,
  "novel_id": "784",
  "novel_pid": null,
  "url": "http://nnv3api.muwai.com/novel/chapter/784/",
}
`;

exports[`support http://nnv4api.dmzj4.com/novel/chapter/784 1`] = `
{
  "chapter_id": null,
  "novel_id": "784",
  "novel_pid": null,
  "url": "http://nnv4api.dmzj4.com/novel/chapter/784",
}
`;

exports[`support http://nnv4api.dmzj4.com/novel/chapter/784.json 1`] = `
{
  "chapter_id": null,
  "novel_id": "784",
  "novel_pid": null,
  "url": "http://nnv4api.dmzj4.com/novel/chapter/784.json",
}
`;

exports[`support http://nnv4api.dmzj4.com/novel/chapter/784/ 1`] = `
{
  "chapter_id": null,
  "novel_id": "784",
  "novel_pid": null,
  "url": "http://nnv4api.dmzj4.com/novel/chapter/784/",
}
`;

exports[`support http://nnv4api.muwai.com/novel/chapter/784 1`] = `
{
  "chapter_id": null,
  "novel_id": "784",
  "novel_pid": null,
  "url": "http://nnv4api.muwai.com/novel/chapter/784",
}
`;

exports[`support http://nnv4api.muwai.com/novel/chapter/784.json 1`] = `
{
  "chapter_id": null,
  "novel_id": "784",
  "novel_pid": null,
  "url": "http://nnv4api.muwai.com/novel/chapter/784.json",
}
`;

exports[`support http://nnv4api.muwai.com/novel/chapter/784/ 1`] = `
{
  "chapter_id": null,
  "novel_id": "784",
  "novel_pid": null,
  "url": "http://nnv4api.muwai.com/novel/chapter/784/",
}
`;

exports[`support http://q.dmzj.com/2476/index.shtml 1`] = `
{
  "chapter_id": undefined,
  "novel_id": "2476",
  "novel_pid": null,
  "url": "http://q.dmzj.com/2476/index.shtml",
  "volume_id": undefined,
}
`;

exports[`support http://v2.api.dmzj.com/novel/2229.json 1`] = `
{
  "chapter_id": null,
  "novel_id": "2229",
  "novel_pid": null,
  "url": "http://v2.api.dmzj.com/novel/2229.json",
}
`;

exports[`support http://v2.api.dmzj.com/novel/download/2229_8303_69136.txt 1`] = `
{
  "chapter_id": "69136",
  "novel_id": "2229",
  "novel_pid": null,
  "url": "http://v2.api.dmzj.com/novel/download/2229_8303_69136.txt",
  "volume_id": "8303",
}
`;
